pkg/service/roommanager.go

```diff
@@ -317,40 +317,48 @@ func (r *RoomManager) getOrCreateRoom(ctx context.Context, roomName livekit.Room
 		return nil, err
 	}
 
-	// construct ice servers
-	room = rtc.NewRoom(ri, *r.rtcConfig, &r.config.Audio, r.telemetry)
-	room.Hold()
+	r.lock.Lock()
 
-	r.telemetry.RoomStarted(ctx, room.Room)
+	currentRoom := r.rooms[roomName]
+	if currentRoom != lastSeenRoom {
+		r.lock.Unlock()
+		currentRoom.Hold()
+		return currentRoom, nil
+	}
 
-	room.OnClose(func() {
-		r.telemetry.RoomEnded(ctx, room.Room)
+	// construct ice servers
+	newRoom := rtc.NewRoom(ri, *r.rtcConfig, &r.config.Audio, r.telemetry)
+
+	newRoom.OnClose(func() {
+		r.telemetry.RoomEnded(ctx, newRoom.Room)
 		if err := r.DeleteRoom(ctx, roomName); err != nil {
 			logger.Errorw("could not delete room", err)
 		}
 
 		logger.Infow("room closed")
 	})
 
-	room.OnMetadataUpdate(func(metadata string) {
-		if err := r.roomStore.StoreRoom(ctx, room.Room); err != nil {
+	newRoom.OnMetadataUpdate(func(metadata string) {
+		if err := r.roomStore.StoreRoom(ctx, newRoom.Room); err != nil {
 			logger.Errorw("could not handle metadata update", err)
 		}
 	})
 
-	room.OnParticipantChanged(func(p types.LocalParticipant) {
+	newRoom.OnParticipantChanged(func(p types.LocalParticipant) {
 		if p.State() != livekit.ParticipantInfo_DISCONNECTED {
 			if err := r.roomStore.StoreParticipant(ctx, roomName, p.ToProto()); err != nil {
 				logger.Errorw("could not handle participant change", err)
 			}
 		}
 	})
 
-	r.lock.Lock()
-	r.rooms[roomName] = room
```

Comment:
we need to put this back

Reply:
Fixed https://github.com/livekit/livekit-server/pull/485/commits/e80fc0852f01722464e5d276ed9fdb9f43448ffd